src/pages/home/sidebar/SidebarLinksData.tsx

```diff
@@ -270,7 +239,8 @@ export default withCurrentReportID(
     withOnyx<SidebarLinksDataProps, SidebarLinksDataOnyxProps>({
         chatReports: {
             key: ONYXKEYS.COLLECTION.REPORT,
-            selector: chatReportSelector as unknown as (report: OnyxEntry<OnyxTypes.Report>) => OnyxCollection<PickedReport & {isUnreadWithMention: boolean}>,
+            // This assertion is needed because the selector in withOnyx expects that the return type will be the same as type in ONYXKEYS but in this case it's not, this is a bug in withOnyx but it's impossible to fix it, when useOnyx will be introduce it will be fixed.
```

Comment:
```suggestion
            // This assertion is needed because the selector in withOnyx expects that the return type will be the same as type in ONYXKEYS but for collection keys the selector is executed for each collection item. This is a bug in withOnyx typings that we don't have a solution yet, when useOnyx hook is introduced it will be fixed.
```

wdyt?